src/Elastic.Apm/Model/Payload/Span.cs

```diff
@@ -62,17 +62,17 @@ public void End()
         {
             if (!Duration.HasValue)
             {
-                this.Duration = (DateTimeOffset.UtcNow - start).TotalMilliseconds;
+                this.Duration = (DateTimeOffset.UtcNow - _start).TotalMilliseconds;
             }
 
-            transaction?.spans.Add(this);
+            Transaction?.SpanCollection.Add(this);
```

Comment:
I renamed `spans` since it conflicted with `Spans` public property. Maybe transaction needs `.RegisterSpan()` so we can encapsulate `Spans` (should that property on `Transaction` be a `IReadOnlyCollection<ISpan>` instead of an array copy?

Reply:
With the Intake V2 API of the apm-server we probably won't need that property. Currently we need that field because within a transaction I send all the spans, but in V2 if I'm not mistaken I can just send a span with it's `transaction_id` separately from the transaction, so we won't need references from a transaction to its spans et all. So let's keep it, and i'll get rid of it in #13.

Reply:
But to also answer the question: yes `IReadOnlyCollection<ISpan>` would make the current situation better.